routing/router.go

```diff
@@ -2848,8 +2848,13 @@ func (r *ChannelRouter) BuildRoute(amt *lnwire.MilliSatoshi,
 	for i, unifier := range unifiers {
 		edge := unifier.getEdge(receiverAmt, bandwidthHints)
 		if edge == nil {
+			fromNode := source
```

Comment:
Can we also add an itest to hit this case? 

Also is this the expected behavior? That the source becomes the default from node if only a single edge was specified?